dpctl/_sycl_queue.pyx

```diff
@@ -338,14 +346,20 @@ cdef DPCTLSyclEventRef _memcpy_impl(
     cdef void *c_dst_ptr = NULL
     cdef void *c_src_ptr = NULL
     cdef DPCTLSyclEventRef ERef = NULL
-    cdef const unsigned char[::1] src_host_buf = None
-    cdef unsigned char[::1] dst_host_buf = None
+    cdef Py_buffer src_buf_view
+    cdef Py_buffer dst_buf_view
+    cdef bint src_is_buf = False
+    cdef bint dst_is_buf = False
+    cdef int ret_code = 0
 
     if isinstance(src, _Memory):
         c_src_ptr = <void*>(<_Memory>src).get_data_ptr()
     elif _is_buffer(src):
-        src_host_buf = src
-        c_src_ptr = <void *>&src_host_buf[0]
+        ret_code = PyObject_GetBuffer(src, &src_buf_view, PyBUF_SIMPLE | PyBUF_ANY_CONTIGUOUS)
+        if ret_code != 0:
```

Comment:
```suggestion
        if ret_code != 0: # pragma: no cover
```

I don't think there's any good way to test these errors, so we can probably just ignore them for the coverage

Same for line 374